### PR TITLE
Repo refactoring

### PR DESCRIPTION
**Pull request by hartkopp**
Hi @marckleinebudde,
this PR contains my idea of a can-tests repo refactoring.
Unfortunately I wasn't able to make 'canpump' compile again (missing Makefile skills).
Does this refactoring fit your expectations?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -91,5 +93,4 @@ clean:
 distclean:
 	rm -f $(PROGRAMS) *.o *~
```
Comment (marckleinebudde) [inline Makefile]: Please add the individual subdirs or `*/*.o`

File: netlayer/test_netlayer.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 #
 # testscript to check CAN filters and CAN frame flow in Linux network layer
 #
-# $Id$
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
 #
```
[the remark this answers is not in the document]
Comment (marckleinebudde) [reply netlayer/test_netlayer.sh]: ...or rather the second line for shell scripts.

**Comment by marckleinebudde** [comment]
`canfdtest` with history: 
https://github.com/marckleinebudde/can-tests/tree/import-canfdtest

File: Makefile [hunk 3]
```diff
@@ -91,5 +93,4 @@ clean:
...
 
-canpump.o:	lib.h
-canpump:	canpump.o	lib.o
+canpump:	raw/canpump.o	lib/lib.o
```
[the remark this answers is not in the document]
Comment (hartkopp) [reply Makefile]: Thanks!

File: netlayer/test_netlayer.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (hartkopp) [reply netlayer/test_netlayer.sh]: Right.

File: Makefile [hunk 4]
```diff
@@ -91,5 +93,4 @@ clean:
 distclean:
 	rm -f $(PROGRAMS) *.o *~
 
-canpump.o:	lib.h
```
[the remark this answers is not in the document]
Comment (hartkopp) [reply Makefile]: Ok. Didn't know better ...

**Comment by hartkopp** [comment]
Moving the canfdtest with history is way better!
Can you please merge that move?
Tnx!

**Comment by marckleinebudde** [comment]
https://github.com/linux-can/can-tests/pull/6

**Pull request closed without merging**